service/matching/matcher.go

```diff
@@ -410,15 +410,9 @@ func (tm *TaskMatcher) emitDispatchLatency(task *internalTask, forwarded bool) {
 		return // should not happen but for safety
 	}
 
-	source := task.source
-	if source == enumsspb.TASK_SOURCE_UNSPECIFIED {
-		// history may not specify the source
-		source = enumsspb.TASK_SOURCE_HISTORY
```

Comment:
I don't have context, is history now expected to provide the source?

Reply:
No this is now handled when creating new internalTask